spanner/session.go

```diff
@@ -42,6 +42,30 @@ import (
 
 const healthCheckIntervalMins = 50
 
+// InactiveTransactionRemovalOptions has configurations for action on long-running transactions.
+type InactiveTransactionRemovalOptions struct {
+	mu sync.Mutex
+	// CloseInactiveTransactions is the configuration to close inactive transactions.
+	CloseInactiveTransactions bool
+	// LogInactiveTransactions is the configuration to log inactive transactions.
+	// Any inactive transaction gets logged only once.
+	LogInactiveTransactions bool
```

Comment:
Here these booleans serve two different purposes and can be independently used,
1. `LogInactiveTransactions` - logs the long running transaction once.
2. `CloseInactiveTransactions` - logs the long running transaction + closes the transaction.

Reply:
I think that the point here is that once you set `CloseInactiveTransactions` to `true`, the value of `LogInactiveTransactions` is ignored. So in that sense they cannot be 'used' independently, and a three-state enum would make more sense (so something like `DoNothing`, `Log` and `LogAndClose`).

Another option could be to really make them independent, and let `CloseInactiveTransactions` skip logging unless `LogInactiveTransactions` has also been set, but I would not be in favor of such a solution.

Reply:
Thanks for the suggestion Arpan and Knut.
I have converted this to an enum now instead of separate boolean fields. `ActionOnInactiveTransaction` will be the public field which is an enum. The default value for this enum is `noAction`. The publicly available enum options are `Warn`, `Close` and `WarnAndClose`